Tests/KituraTests/TestDecodingErrorExtension.swift

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright IBM Corporation 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import XCTest
+
+class TestDecodingErrorExtension: XCTestCase {
+    
+    static var allTests: [(String, (TestDecodingErrorExtension) -> () throws -> Void)] {
+        return [
+            ("testMalformedJson", testMalformedJson),
+            ("testMissingRequiredKey", testMissingRequiredKey),
+            ("testWrongType", testWrongType),
+            ("testMissingValue", testMissingValue),
+            
+        ]
+    }
+    
+    func testMalformedJson() {
+        
+        let testJSON = """
+{
+"name": "woof", THISSHOULDNOTBEHERE
+"value": 3,
+"subCodable": {
+  "subName": "subWoof",
+  "subSubTest": {
+    "subSubSubInt" : 4
+    }
+  }
+}
+"""
+        do {
+            _ = try JSONDecoder().decode(TestCodable.self, from: testJSON.data(using: .utf8)!)
+            XCTFail("We should have had a decoding error, but had no error.")
+        } catch  {
+            if let decodingError = error as? DecodingError {
+                XCTAssert(decodingError.humanReadableDescription.contains("The JSON appears to be malformed."), "DecodingError.humanReadableDescription not what we expected.")
+            } else {
+                // Linux Swift does not throw a DecodingError for malformed JSON, an Error "The operation could not be completed"
+                #if !os(Linux)
```

Comment:
The only issue I can see here is that in the future, if Linux changes to be consistent with Darwin, we'll fail to make this assertion.  Since we don't know if/when this may be addressed, we can't add a Swift version check (well, we could, but it would be along the lines of `#if !os(Linux) || swift(>4.1)` and we'd have to bump the version each time there was another release that didn't fix it).  We should remember to revisit this in the future.